net/pfSense-pkg-freeradius3/files/usr/local/pkg/freeradius.inc

```diff
@@ -1004,7 +1004,8 @@ function freeradius_clients_resync($restart_svc = true) {
 	if (!empty($arrclients)) {
 		foreach ($arrclients as $item) {
 			$varclientip = $item['varclientip'];
-			$varclientsharedsecret = $item['varclientsharedsecret'];
+			// Bug #7836: Add single quotes to avoid breaking configuration with certain chars in secret
+			$varclientsharedsecret = "'" . $item['varclientsharedsecret'] . "'";
```

Comment:
That is not doable because it escapes | as well, which won't be un-escaped. Users really will need to take care of their whacky shared secrets themselves.